test/unit/test_engagement_manager.py

```diff
@@ -60,6 +60,45 @@ def test_get_engagement__reuses_engagement_when_cached_engagement_unexpired(
     assert engagement is cached_engagement
 
 
+def test_get_engagement__using_endpoint_id(client_configuration: QCSClientConfiguration, httpx_mock: HTTPXMock):
```

Comment:
This test doesn't cover retrieving a cached endpoint-specific engagement. Rather than writing a new test, you might consider removing this and just adding a second engagement for your case to the two tests above: `test_get_engagement__refreshes_engagement_when_cached_engagement_expired` and `test_get_engagement__reuses_engagement_when_cached_engagement_unexpired`

To do this, I'd recommend augmenting the `cache_engagement()` test helper to include an optional endpoint ID

Reply:
(`cache_engagement()` implicitly tests caching an engagement for the first time, as you're doing here)